### PR TITLE
fix: otlp metrics proto type error

### PR DESCRIPTION
**Pull request by iutx**
#### What this PR does / why we need it:
1. Delete the OpenTelemetry proto files.
2. Introducing `proto_fetcher.sh` to solve the issue of importing external proto files.
3. Introducing mirror params, If users are unable to access GitHub, they can solve this by introducing a mirror. e.g.
```shell
 OTEL_PROTO_REPO_MIRROR=https://gitee.com/rocket-mirror/opentelemetry-proto.git make proto-go-in-local 
```
It has been tested on both Linux and macOS.
The workflow verification results are as follows:
image

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](paste your link here)

#### Specified Reviewers:

/assign @sfwn @sixther-dc 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English | fix otlp metrics proto type error             |
| 🇨🇳 中文    |    修复 otlp metrics proto 类型错误          |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: api/proto/oap/collector/receiver/opentelemetry/opentelemetry.proto [hunk 1]
```diff
@@ -23,7 +22,7 @@ message PostMetricsServiceRequest {
   // element. Intermediary nodes (such as OpenTelemetry Collector) that receive
   // data from multiple origins typically batch the data before forwarding further and
   // in that case this array will contain multiple elements.
-  repeated erda.oap.opentelemetry.ResourceMetrics resource_metrics = 1;
+  repeated erda.oap.opentelemetry.metrics.v1.ResourceMetrics resource_metrics = 1;
```
Comment (sfwn) [inline api/proto/oap/collector/receiver/opentelemetry/opentelemetry.proto:25]: Can we remove the prefix "erda.oap"? Maybe we can use otel proto directly.

**Review by sfwn — CHANGES_REQUESTED** [review]
Add a cmd in Makefile of proto-api, to clarify the way how we intergrate these otel proto with erda proto.

File: api/proto/oap/collector/receiver/opentelemetry/opentelemetry.proto (hunk 1 above)
Comment (iutx) [reply api/proto/oap/collector/receiver/opentelemetry/opentelemetry.proto]: Done.
Introducing `proto_fetcher.sh` to solve the issue of importing external proto files.

The command remains as "make proto-go-in-local".

If users are unable to access GitHub, they can solve this by introducing a mirror. e.g.
```shell
 OTEL_PROTO_REPO_MIRROR=https://gitee.com/rocket-mirror/opentelemetry-proto.git make proto-go-in-local 
```

File: api/proto/oap/collector/receiver/opentelemetry/opentelemetry.proto [hunk 2]
```diff
@@ -23,7 +22,7 @@ message PostMetricsServiceRequest {
   // element. Intermediary nodes (such as OpenTelemetry Collector) that receive
   // data from multiple origins typically batch the data before forwarding further and
   // in that case this array will contain multiple elements.
-  repeated erda.oap.opentelemetry.ResourceMetrics resource_metrics = 1;
+  repeated .opentelemetry.proto.metrics.v1.ResourceMetrics resource_metrics = 1;
```
Comment (sfwn) [inline api/proto/oap/collector/receiver/opentelemetry/opentelemetry.proto:25]: strange prefix `.`

File: api/proto/proto_fetcher.sh [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+  target_path="./opentelemetry"
+
+  # cleanup
+  echo "cleanup opentelemetry proto files"
+  rm -rf $target_path
+
+  # fetch code
+  echo -e "\n#1 [otel] fetch code:"
+  fetch_code --repo=https://github.com/open-telemetry/opentelemetry-proto.git \
+             --commit-id=395c8422fe90080314c7d9b4114d701a0c049e1f \
```
Comment (sfwn) [inline api/proto/proto_fetcher.sh:122]: extract this commit-id to a variable and put it at the beginning of the file.

**Review by sfwn — CHANGES_REQUESTED** [review]

File: api/proto/oap/collector/receiver/opentelemetry/opentelemetry.proto (hunk 2 above)
Comment (iutx) [reply api/proto/oap/collector/receiver/opentelemetry/opentelemetry.proto:25]: The current package name is "opentelemetry", and the outermost package with the same name cannot be directly referenced.

File: api/proto/proto_fetcher.sh (hunk 3 above)
Comment (iutx) [reply api/proto/proto_fetcher.sh]: done

**Review by sfwn — APPROVED** [review]

**Pull request merged**